tests/phpunit/integration/Core/User_Surveys/REST_User_Surveys_ControllerTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\User_Surveys\REST_User_Surveys_ControllerTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\User_Surveys
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\User_Surveys;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\User_Surveys\REST_User_Surveys_Controller;
+use Google\Site_Kit\Tests\TestCase;
+
+class REST_User_Surveys_ControllerTest extends TestCase {
+
+	/**
+	 * Authentication object.
+	 *
+	 * @var Authentication
+	 */
+	private $authentication;
+
+	public function setUp() {
+		parent::setUp();
+		$context              = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->authentication = new Authentication( $context );
+	}
+
+	public function test_register() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+
+		$controller = new REST_User_Surveys_Controller( $this->authentication );
+		$controller->register();
+
+		$this->assertTrue( has_filter( 'googlesitekit_rest_routes' ) );
+	}
+
```

Comment:
This is a good start but let's add at least one test for each route, similar to `REST_Feature_Tours_ControllerTest`.

We don't need to worry about requests hitting the proxy since these will be blocked by `WP_HTTP_BLOCK_EXTERNAL`. We can assume that if the request responds with a `http_request_not_executed` error, then it was successfully sent to the proxy. It would be good to assert that the request was made with the expected args in the body as well.

Reply:
I have added tests for Google_Proxy methods that test requests to the proxy. It should be enough I guess, shouldn't it?

Reply:
Thank you for adding those for `Google_Proxy` 👍 

I would prefer if we had tests that actually invoke the registered routes as more of an end-to-end test although we wouldn't have to duplicate everything that's covered by the other tests you added.

If you think this would take too much time we can skip it in this case but I think we should try to add better coverage for our REST routes going forward.

Reply:
Added a test that checks the registered route arguments. Testing REST route handlers themself is quite cumbersome and will take too much time to do it properly. Let's skip it.